modules/administration-guide/pages/default-extensions-for-microsoft-visual-studio-code.adoc

```diff
@@ -0,0 +1,171 @@
+
+[NOTE]
+====
+It is possible to specify several extensions separated by semicolon.
+
+[source,yaml]
+----
+  DEFAULT_EXTENSIONS='/projects/extension-1.vsix;/projects/extension-2.vsix'
+----
+====
+
+Below you can find several examples how to add *.vsix* files to your workspace and how to define the DEFAULT_EXTENSIONS environment variable.
+
+.Procedure
+
+This documentation describes three different ways to embed default .vsix extensions for your workspace.
+
+* Put the extension binary to the source repository.
++
+The easiest way is to put the extension binary to the Git repository and define the environment variable in the devfile.
+If the *extension.vsix* file exists in the repository root, the __DEFAULT_EXTENSIONS__ could be set for a tooling container.
+Just specify it in your *.devfile.yaml* like below.
++
+====
+[source,yaml]
+----
+schemaVersion: 2.3.0
+metadata:
+  generateName: example-project
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      env:
+        - name: 'DEFAULT_EXTENSIONS'
+          value: '/projects/example-project/extension.vsix'
+----
+====
+
+* Use the Devfile *postStart* event to fetch extension binaries from the network.
++
+It is possible to use *curl* or *wget* to download extensions to your workspace.
+For that you need to:
++
+--
+** specify a devfile command to download one or several extensions to your workpace
+** add a *postStart* event to run the command on workspace startup
+** define the __DEFAULT_EXTENSIONS__ environment variable in the Devfile
+--
++
+The following sample demonstrates what should be added to the devfile to add two extensions.
++
+====
+[source,yaml]
+----
+schemaVersion: 2.3.0
+metadata:
+  generateName: example-project
+components:
+  - name: tools
+    container:
+      image: quay.io/devfile/universal-developer-image:ubi8-latest
+      env:
+        - name: DEFAULT_EXTENSIONS
+          value: '/tmp/extension-1.vsix;/tmp/extension-2.vsix'
+
+commands:
+  - id: add-default-extensions
+    exec:
+      # name of the tooling container
+      component: tools
+      # download several extensions using curl
+      commandLine: |
+        curl https://.../extension-1.vsix --location -o /tmp/extension-1.vsix
+        curl https://.../extension-2.vsix --location -o /tmp/extension-2.vsix
+
+events:
+  postStart:
+    - add-default-extensions
+----
+====
++
+[WARNING]
+====
+It some cases *curl* may download *gzip* compressed file. This may make it impossible to install the extension.
+To fix that try to save the file to *.vsix.gz* and then decompress it with *gunzip*. It will replace *.vsix.gz* with unpacked *.vsix* file.
```

Comment:
```suggestion
In some cases *curl* may download a *gzip* compressed file. This may make it impossible to install the extension.
To fix that try to save the file as a *.vsix.gz* file and then decompress it with *gunzip*. This will replace the *.vsix.gz* file with an unpacked *.vsix* file.
```